3.8/graphs-custom-pregel.md

```diff
@@ -0,0 +1,1511 @@
+- `false?` returns true if `expr` is considered false, returns true otherwise.
+- `not` is an alias for `false?`.
+
+```js
+> ["true?", 5]
+ = true
+> ["true?", 0]
+ = true
+> ["true?", false]
+ = false
+> ["true?", "Hello world!"]
+ = true
+> ["false?", 5]
+ = false
+```
+
+#### Comparison operators
+
+_compares on value to other values_
+
+```js
+["eq?", proto, expr...]
+["gt?", proto, expr...]
+["ge?", proto, expr...]
+["le?", proto, expr...]
+["lt?", proto, expr...]
+["ne?", proto, expr...]
+```
+
+Compares `proto` to all other expressions according to the selected operator.
+Returns true if all comparisons are true. Returns true for the empty list.
+Relational operators are only available for numeric values. If proto is a
+boolean value the other values are first converted to booleans using `true?`,
+i.e. you compare their truthiness.
+
+The operator names translate as follows:
+- `eq?` -- `["eq?", left, right]` evaluates to `true` if `left` is equal to `right`
+- `gt?` -- `["gt?", left, right]` evaluates to `true` if `left` is greater than `right`
+- `ge?` -- `["ge?", left, right]` evaluates to `true` if `left` is greater than or equal to`right`
+- `le?` -- `["le?", left, right]` evaluates to `true` if `left` is less than or equal to `right`
+- `lt?` -- `["lt?", left, right]` evaluates to `true` if `left` is less than `right`
+- `ne?` -- `["ne?", left, right]` evaluates to `true` if `left` is not equal to `right`
+
+Given more than two parameters
+
+```js
+[<op>, proto, expr_1, expr_2, ...]
+```
+
+is equivalent to
+
+```js
+["and", [<op>, proto, expr_1], [<op>, proto, expr_2], ...]
+```
+
+except that `proto` is only evaluated once.
+
+```js
+> ["eq?", "foo", "foo"]
+ = true
+> ["lt?", 1, 2, 3]
+ = true
+> ["lt?", 1, 3, 0]
+ = false
+> ["ne?", "foo", "bar"]
+ = false
+```
+
+#### Lists
+
+_sequential container of inhomogeneous values_
+
+```js
+["list", expr...]
+["list-cat", lists...]
+["list-append", list, expr...]
+["list-ref", list, index]
+["list-set", arr, index, value]
+["list-empty?", value]
+["list-length", list]
+```
+
+`list` constructs a new list using the evaluated `expr`s. `list-cat`
+concatenates given lists. `list-append` returns a new list, consisting of the
+old list and the evaluated `expr`s. `list-ref` returns the value at `index`.
+Accessing out of bound is an error. Offsets are zero based. `list-set`
+returns a copy of the old list, where the entry and index `index` is replaced
+by `value`. Writing a index that is out of bounds is an error. `list-empty?`
+returns true if and only if the given value is an empty list. `list-length`
+returns the length of the list.
+
+#### Sort
+
+_sort a list_
+
+```js
+["sort", compare, list]
+`sort` sorts a list by using the compare function. `compare` is called with
+two parameters `a` and `b`. `a` is considered less than `b` is the return value
+of this call is considered true. The sort is **not** stable.
```

Comment:
It should be noted that the resulting order is ascending, based on the compare function.